tests/MenderAPI/authentication.py

```diff
@@ -17,20 +17,54 @@
 
 class Authentication:
     auth_header = None
-    email = "admin@admin.net"
-    password = "averyverystrongpasswordthatyouwillneverguess!haha!"
+
+    email = ""
+    password = ""
+
+    multitenancy = False
+    current_tenant = {}
+
+    def __init__(self, username="admin@admin.net", password="averyverystrongpasswordthatyouwillneverguess!haha!"):
+        self.email = username
+        self.password = password
+
+    def set_tenant(self, username, password):
+        return self.new_tenant(username, password)
+
+    def new_tenant(self, username, password):
+        self.multitenancy = True
+        self.reset_auth_token()
+        self.email = username
+        self.password = password
+        self.get_auth_token()
 
     def get_auth_token(self):
         if self.auth_header is not None:
             return self.auth_header
 
-
         # try login - the user might be in a shared db already (if not running xdist)
         r = self._do_login(self.email, self.password)
 
         # ...if not, create user
         if r.status_code is not 200:
-            self._create_user(self.email, self.password)
+            if self.multitenancy:
+                tenant_id = self._create_tenant(self.email)
+                tenant_id = tenant_id.strip()
+
+                self._create_tenant_user(self.email, tenant_id)
+                self._create_user(self.email, self.password, tenant_id)
+
+                tenant_data = self._get_tenant_admin(tenant_id)
+                logging.info("(%s) tenant_key is: %s" % (self.email, tenant_data))
```

Comment:
iirc `_get_tenant_admin` returns both `id` and `tenant_token`, so just to give the devil it's due let's change the message here.
also let's standardize on `tenant_token` - different names float around jira etc, but this is the correct one.